docs/guide/interactions/modals.rst

```diff
@@ -0,0 +1,143 @@
+
+.. warning:: 
+
+    Ensure that class-attribute defined fields do not conflict with attributes or methods defined in the :class:`parent class <discord.ui.Modal>`, 
+    as this will cause unexpected behaviour.
+
+
+You can customize individual modal instances with normal attribute access and assignment.
+For example, to change the label of the ``name`` field to ``'Your Name'`` you can do the following:
+
+.. code-block:: python
+
+    form = FeedbackForm()
+    form.name.label = 'Your Name'
+
+
+Handling User Responses
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once the user has clicked `Submit` the :meth:`Modal.on_submit <discord.ui.Modal.on_submit>` callback is called. 
+The callback is passed a new :class:`~discord.Interaction` instance, which requires a response.
+
+In the example above we respond by sending a message to the user confirming their submission was recorded.
+
+
+Handling an error
+~~~~~~~~~~~~~~~~~~
+
+The :meth:`Modal.on_error <discord.ui.Modal.on_error>` method is called when an exception is raised
+within :meth:`Modal.on_submit <discord.ui.Modal.on_submit>`, and can be used to handle an error or respond to the user.
+
+This method is passed the exception raised, and the :class:`~discord.Interaction` instance, allowing you to handle the error and, 
+if necessary, send a response to the user.
+
+
+.. code-block:: python
+    :emphasize-lines: 13-15
+
+    class FeedbackForm(discord.ui.Modal, title='Feedback'):
+        name = discord.ui.TextInput(label='Name')
+        feedback = discord.ui.TextInput(label='Feedback', style=discord.TextStyle.long)
+        additional_information = discord.ui.TextInput(
+            label='Additional Information', style=discord.TextStyle.long, required=False
+        )
+
+        async def on_submit(self, interaction: discord.Interaction) -> None:
+            ... # do something with the data
+
+            await interaction.response.send_message(f'Thank you {self.name.value}, your submission was recorded.')
+
+        async def on_error(self, error: Exception, interaction: discord.Interaction) -> None:
+            if not interaction.response.is_done():
+                await interaction.response.send_message('An error occurred, please try again.')
+
+
+Sending a Modal
+-----------------
+
+A modal can be sent by calling the :meth:`InteractionResponse.send_modal` method when handling an :class:`Interaction`.
+
+For example, you can respond with a modal when somebody uses a slash command:
+
+
+.. code-block:: python
+
+    @discord.app_commands.command()
+    async def feedback(interaction: discord.Interaction) -> None:
+        """Send important feedback."""
+        await interaction.response.send_modal(FeedbackForm())
+
+
+Once a modal is sent, a user has the option to submit values. These should be handled either in an implementation of the
+:meth:`~discord.ui.Modal.on_submit` method or via :meth:`Modal.wait() <discord.ui.Modal.wait>`,
```

Comment:
Kinda on the fence about including `Modal.wait` usage as I've never seen code that uses the coro

Reply:
Well if that works, then it's fine to leave it there